### ISSUE TITLE
Fail to build docs w/ Python < 3.5.2

### ISSUE THREAD

**Issue by zaufi** [sphinx-doc/sphinx#5492]
### Problem
Using CentOS 7 and Python 3.5.1 available via [Software Collections](https://www.softwarecollections.org/en/scls/rhscl/rh-python35/) I've got a build failure w/ Sphinx >= 1.8.0:

~~~
Traceback (most recent call last):
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/registry.py", line 472, in load_extension
    mod = __import__(extname, None, None, ['setup'])
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/transforms/post_transforms/compat.py", line 13, in <module>
    from typing import TYPE_CHECKING
ImportError: cannot import name 'TYPE_CHECKING'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/application.py", line 224, in __init__
    self.setup_extension(extension)
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/application.py", line 449, in setup_extension
    self.registry.load_extension(self, extname)
  File "/opt/scl/ci/lib/python3.5/site-packages/sphinx/registry.py", line 475, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname, err)
sphinx.errors.ExtensionError: Could not import extension sphinx.transforms.post_transforms.compat (exception: cannot import name 'TYPE_CHECKING')
~~~

### Environment info
- OS: CentOS 7
- Python version: 3.5.1
- Sphinx version: 1.8.0 / 1.8.1

**Comment by carlosalbaladejo** [comment]
Same problem on virtual environment with Python version: 3.5.1.

> sphinx.errors.ExtensionError: Could not import extension sphinx.transforms.post_transforms.compat (exception: cannot import name 'TYPE_CHECKING')

TYPE_CHECKING is not defined in `typing.py` library for Python 3.5.
I solved it upgrading to Python 3.7

**Comment by zaufi** [comment]
Yep, I know that `TYPE_CHECKING` is available since 3.5.2. But,

- I can't upgrade: Python 3.5.2 just doesn't exist in SCLs -- only 3.5.1,
 and using SCLs is the only way for me to get Python 3;
- that is a weird thing in Sphinx Python compatibility -- supported
 versions are: Python < 3.5 (3.4 supported via separate package) and Python
 >= 3.5.2 (w/ native support)... I don't see in the docs that we have an
 unsupported versions hole [3.5.0 ..3.5.2)  %)

**Comment by tk0miya** [comment]
Fixed by #5510.
Thank you for reporting.
